spec/steps/other_info/form_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+RSpec.describe Steps::OtherInfoForm do
+  let(:form) { described_class.new(application:, **arguments) }
+
+  let(:arguments) do
+    {
+      application:,
+      other_info:,
+      conclusion:,
+      concluded:,
+    }
+  end
+
+  let(:application) { instance_double(Claim, update!: true) }
+  let(:other_info) { 'other relevent information' }
+  let(:conclusion) { 'conclusion' }
+  let(:concluded) { 'yes' }
+
+  describe '#save' do
+    context 'when all fields are set' do
+      it 'is valid' do
+        expect(form.save).to be_truthy
+        expect(application).to have_received(:update!)
+        expect(form).to be_valid
+      end
+    end
+  end
+
+  describe '#valid?' do
+    context 'when all fields are set' do
+      it 'is valid' do
+        expect(form).to be_valid
+      end
+    end
```

Comment:
we need to be testing the other validation criteria, expecially for something like `conclusion` which needs conditional validation and `concluded` which needs to have value in a specific set.

Reply:
will do

Reply:
hopefully resolved for conditionals

Reply:
So the issue with the tests here is that they only ever test for when the input is valid. we should also have tests for when the input is not valid..


i.e.
Scenario 1:
concluded = 'other' 
result => is invalid

Scenario 2:
concluded = 'yes' 
conclusion = nil
result => is invalid
 
Scenario 3:
concluded = 'no' 
conclusion = 'some text'
result => is valid, but conclusion is NOT saved to DB